unittest/test_Util.cpp

```diff
@@ -21,8 +21,12 @@
 #include "../src/Util.hpp"
 #include "TestUtil.hpp"
 
+#include <ostream>
+//
```

Comment:
Why does `ostream` need to be included before `third_party/doctest.h`?

Reply:
I made a cleaner fix for this by defining `DOCTEST_CONFIG_USE_STD_HEADERS` in `TestUtil.hpp`.

Reply:
Why? I know it's cleaner and more standard compliant to use DOCTEST_CONFIG_USE_STD_HEADERS . But doctest has it off by default intentionally since it's faster to not include but forward declare stuff. And doctest runs tests on all MSVC versions.

And even when I add that define I do not need ostream (on clang and gcc). Why does MSVC need ostream?

Reply:
I timed 3 runs of building the tests on linux with gcc 9.3 without and with the option, on average it increased the real time by `0.38` seconds and the user time by `0.79` seconds, so there is a difference.

I suspect the reason MSVC needs ostream is the same as why I had to add some other standard headers like algorithm, gcc implicitly includes or defines some things from other headers when you include one and MSVC does not.

Without the option I get a big page full of errors, the first of which is:

```none
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\xstring(426): error C2027: use of undefined type 'std::basic_ostream<char,std::char_traits<char>>'
```

Trying the latest version of doctest did not fix the problem either.

I changed this to set the option **only** for MSVC.

Reply:
Shouldn't [<iofwd>](https://en.cppreference.com/w/cpp/header/iosfwd) be used instead of `ofstream`?  

Forward declarations shouldn't penalize that much as definitions.

Reply:
@cristianadam I tried this, but it seems MSVC requires `<ostream>` specifically.

Reply:
Sorry, now I see it was `test_Util.cpp` and not a header file. 

Being in a test file it shouldn't matter that much, right? One can have a precompiled header and it shouldn't be a problem.

Reply:
The issue with headers and doctest is increasing the compile time of test sources, which can be bothersome when you're doing repeated edit/compile/run cycles to run unit tests. I measured the difference for this particular configuration option, and it is fairly minimal, but regardless I changed the option to only be enabled for MSVC.